cpp/arcticdb/column_store/test/benchmark_memory_segment.cpp

```diff
@@ -0,0 +1,104 @@
+/* Copyright 2023 Man Group Operations Limited
+ *
+ * Use of this software is governed by the Business Source License 1.1 included in the file licenses/BSL.txt.
+ *
+ * As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+ */
+
+#include <benchmark/benchmark.h>
+
+#include <arcticdb/column_store/memory_segment.hpp>
+#include <arcticdb/stream/test/stream_test_common.hpp>
+#include <folly/container/Enumerate.h>
+
+#include <algorithm>
+
+using namespace arcticdb;
+
+// run like: --benchmark_time_unit=ms --benchmark_filter=.* --benchmark_min_time=5x
+
+std::vector<bool> get_sparse_bits(size_t num_rows, size_t num_set){
+    auto sparse_bits = std::vector<bool>(num_rows, false);
+    std::fill(sparse_bits.begin(), sparse_bits.begin()+num_set, true);
+    std::random_device rd;
+    std::mt19937 g(rd());
+    std::shuffle(sparse_bits.begin(), sparse_bits.end(), g);
+    return sparse_bits;
+}
+
+std::vector<uint64_t> get_random_permutation(size_t num_rows){
+    auto result = std::vector<uint64_t>(num_rows);
+    std::iota(result.begin(), result.end(), 1);
+    std::random_device rd;
+    std::mt19937 g(rd());
+    std::shuffle(result.begin(), result.end(), g);
+    return result;
+}
+
+SegmentInMemory get_shuffled_segment(const StreamId &id, size_t num_rows, size_t num_columns, std::optional<float> non_index_sparse_percent = std::nullopt){
+    auto fields = std::vector<FieldRef>(num_columns);
+    for (auto i=0u; i<num_columns; ++i){
+        fields[i] = scalar_field(DataType::UINT64, "column_"+std::to_string(i));
+    }
+    auto segment = SegmentInMemory{
+        get_test_descriptor<stream::TimeseriesIndex>(id, fields),
+        num_rows,
+        false,
+        non_index_sparse_percent.has_value()
+    };
+
+    for (auto i=0u; i<=num_columns; ++i){
+        auto& column = segment.column(i);
+        auto values = get_random_permutation(num_rows);
+        // We ensure the column we're sorting by is NOT sparse. As of 2023/12 sorting by sparse columns is not supported.
+        auto num_set = non_index_sparse_percent.has_value() && i!=0 ?
```

Comment:
There's a lot going on with this ternary operator, I'd prefer exploding this out to if-else conditions